jwst/tso_photometry/tso_photometry.py

```diff
@@ -54,14 +60,30 @@ def tso_aperture_photometry(datamodel, xcenter, ycenter, radius, radius_inner, r
         phot_aper = CircularAperture((xcenter, ycenter), r=radius)
         bkg_aper = CircularAnnulus((xcenter, ycenter), r_in=radius_inner, r_out=radius_outer)
 
-    # convert the input data and errors from MJy/sr to Jy
-    if datamodel.meta.bunit_data != "MJy/sr":
-        raise ValueError("data is expected to be in units of MJy/sr")
-    factor = 1.0e6 * datamodel.meta.photometry.pixelarea_steradians
-    datamodel.data *= factor
-    datamodel.err *= factor
-    datamodel.meta.bunit_data = "Jy"
-    datamodel.meta.bunit_err = "Jy"
+    if datamodel.meta.bunit_data == "MJy/sr":
+        # Convert the input data and errors from MJy/sr to Jy
+        factor = 1e6 * datamodel.meta.photometry.pixelarea_steradians
+        datamodel.data *= factor
+        datamodel.err *= factor
+        datamodel.meta.bunit_data = "Jy"
+        datamodel.meta.bunit_err = "Jy"
+    elif datamodel.meta.bunit_data == "DN/s":
+        # Convert the input data and errors from DN/s to electrons
+        factor = datamodel.meta.exposure.integration_time * gain_model.data
```

Comment:
Are you sure you mean EFFINTTM (datamodel.meta.exposure.integration_time)?  I think that makes sense, but please check https://jwst-docs.stsci.edu/accessing-jwst-data/jwst-science-data-overview/jwst-time-definitions

We should probably also make clear in the documentation that these aren't 'true' electrons but idealized electrons; i.e. they've been flatfielded, corrected for nonlinearity, etc.

Reply:
Yes, we want the total time spent collecting photons per integration, so EFFINTTM. Fair point about the documentation note though

Reply:
After the rebasing is properly handled (sorry for totally messing that up), noting this in the documentation is the only edit request that should remain. Where were you thinking this docs change should be made - just within the comments/docstring for this code, or somewhere else?

Reply:
I'd be inclined to add a line or two to the 'Assumptions' section of https://github.com/spacetelescope/jwst/blob/main/docs/jwst/tso_photometry/description.rst
I.e., input is usually assumed to be flux calibrated in units of MJy/sr, in which case the output will be in Jy.  If flux calibrated was skipped, it will convert from DN/s to flatfielded e-/sec.